sympy/polys/monomials.py

```diff
@@ -255,24 +255,38 @@ def monomial_pow(A, n):
     """Return the n-th pow of the monomial. """
     return tuple([ a*n for a in A ])
 
-def monomial_gcd(A, B):
+def monomial_gcd(*args):
     """
-    Greatest common divisor of tuples representing monomials.
+    Computes the greatest common divisor (GCD) of the exponents for each
+    variable in the monomials.
 
-    Examples
-    ========
+    Parameters
+    ==========
 
-    Lets compute GCD of `x*y**4*z` and `x**3*y**2`::
+    *args : tuple of tuples or multiple tuples
+        If a single argument is provided, it should be a tuple of tuples
+        representing monomials.
+        If two arguments are provided, they should be two tuples representing
+        monomials.
 
-        >>> from sympy.polys.monomials import monomial_gcd
+    Returns
+    =======
 
-        >>> monomial_gcd((1, 4, 1), (3, 2, 0))
-        (1, 2, 0)
-
-    which gives `x*y**2`.
+    tuple
+        The greatest common divisor of the exponents for each variable in the
+        monomials.
 
     """
-    return tuple([ min(a, b) for a, b in zip(A, B) ])
+    if len(args) == 1:
+        monomials = args[0]
+        monomial_gcd = tuple(map(min, zip(*monomials)))
+    elif len(args) == 2:
+        A, B = args
+        monomial_gcd = tuple([min(a, b) for a, b in zip(A, B)])
```

Comment:
Is it possible to costruct the tuple directly without a list?

Reply:
There is no tuple comprehension that can construct a tuple directly. It is possible to construct a tuple from a generator rather than using a list though. That can be slower though:
```python
In [1]: A = (1,2,3,4,5)

In [2]: B = (4,5,6,7,7)

In [3]: %timeit tuple(min(a, b) for a, b in zip(A, B))
2.5 µs ± 47.1 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [4]: %timeit tuple([min(a, b) for a, b in zip(A, B)])
2.1 µs ± 30.4 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```
The reason that constructing a list is faster is because it is done in a single function call e.g. the code
```python
stuff = tuple([f(x) for x in things])
```
translates to (the interpreter literally does this translation):
```python
def _comp():
    _stuff = []
    for x in things:
        _stuff.append(f(x))

stuff = tuple(_comp())
```
This has a single function call that does the whole loop and then the list is converted (very efficiently) to a tuple. On the other hand this
```python
stuff = tuple(f(x) for x in things)
```
translates to
```python
def _comp():
    for x in things:
        yield f(x)

_gen = _comp()

stuff = tuple(_gen)
```
Now the tuple constructor has to loop over the generator. For each item the generator function frame has to be resumed in order to execute the code until the next yield. What this means in effect is that we have the overhead of a function call for each element in the list rather than a single function call for the whole list.

Generator expressions are more efficient than list comprehensions if there is any possibility to break out of the loop with an early return e.g. `any(f(x) for x in stuff)` might only loop over the first item in stuff when stuff might be a large list of items whereas `any([f(x) for x in stuff])` would always loop over the whole list.

Reply:
Interesting, thanks for the information.